### PR TITLE
feat(titus): Support specifying a percentage of instances to relaunch

### PR DESCRIPTION
**Pull request by ajordens**

### CODE REVIEW

**Comment by ajordens** [comment]
image

File: app/scripts/modules/titus/src/migration/titusMigrationConfigurer.component.html [hunk 1]
```diff
@@ -10,11 +10,21 @@
     <div class="config-field">
       <label>
         Batch size
-        <help-field key="strategy.rollingPush.concurrentRelaunches"></help-field>
+        <help-field key="strategy.rollingPush.concurrentRelaunches.migration"></help-field>
       </label>
       <input class="form-control input-sm" type="number" min="1"
              ng-model="$ctrl.config.config.concurrentRelaunches"
              ng-change="$ctrl.configChanged()" />
+      <ui-select ng-model="$ctrl.config.config.concurrentRelaunchesAsPercentage"
```
Comment (ajordens) [inline app/scripts/modules/titus/src/migration/titusMigrationConfigurer.component.html]: Almost certainly a more elegant way of doing this

File: app/scripts/modules/titus/src/migration/titusMigrationConfigurer.component.html [hunk 2]
```diff
@@ -23,7 +33,7 @@
         </label>
         <input class="form-control input-sm" type="number" min="0"
                ng-model="$ctrl.config.config.waitTime"
-               ng-change="$ctrl.configChanged()"/> seconds between updates
+               ng-change="$ctrl.configChanged()"/> <span style="font-size: 12px">seconds between updates</span>
```
Comment (ajordens) [inline app/scripts/modules/titus/src/migration/titusMigrationConfigurer.component.html]: image

Quick hack to fix this wrapping?

**Comment by sargun** [comment]
Does it make sense to add a validator ensuring the percentage is > 0 and =< 100

**Comment by ajordens** [comment]
Fair point @sargun.

@anotherchrisberry -- You probably have something better than `max="{{ $ctrl.config.config.concurrentRelaunchesAsPercentage ? 100 : 9999 }}"`

image

JavaScript++

**Pull request merged**